libusb/descriptor.c

```diff
@@ -212,7 +216,19 @@ static int parse_interface(libusb_context *ctx,
 		usb_interface->altsetting = altsetting;
 
 		ifp = altsetting + usb_interface->num_altsetting;
+
+		usb_interface->num_altsetting++;
+
+		ifp->endpoint = NULL;
+		ifp->extra = NULL;
+		ifp->extra_length = 0;
+
 		parse_descriptor(buffer, "bbbbbbbbb", ifp);
+
+		// stash in case we have to clear it
+		uint8_t stashedNumEndpoints = ifp->bNumEndpoints;
+		ifp->bNumEndpoints = 0;
```

Comment:
Stashing this ensures that we clear only the endpoints that are actually instantiated.

Reply:
Line 228 says: stash /in case/ we have to clear it. Then line 231 goes on to clear it. This is confusing.

> bNumEndpoints will represent the number of *initialised* endpoints

Can't we call it e.g. numInitialisedEndpoints then, instead of "stashedNumEndpoints"?